admin/server/github.go

```diff
@@ -30,14 +30,23 @@ func (s *Server) GetGithubRepoStatus(ctx context.Context, req *adminv1.GetGithub
 
 	// If the user has not granted access, return instructions for granting access
 	if !ok {
-		grantAccessURL, err := url.JoinPath(s.opts.FrontendURL, "/github/connect")
+		grantAccess, err := url.JoinPath(s.opts.FrontendURL, "/github/connect")
 		if err != nil {
 			return nil, status.Errorf(codes.Internal, "failed to create redirect URL: %s", err)
 		}
 
+		grantAccessURL, err := url.Parse(grantAccess)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "failed to create redirect URL: %s", err)
+		}
+
+		qry := grantAccessURL.Query()
+		qry.Set("auto_redirect", "true")
+		grantAccessURL.RawQuery = qry.Encode()
```

Comment:
Yeah that seems neat. Although as part of another PR I changed it slightly so that all these paths are created at startup itself so that we don't need to create it on every request.